src/plugins/data/common/data_frames/utils.ts

```diff
@@ -52,9 +53,30 @@ export const convertResult = (response: IDataFrameResponse): SearchResponse<any>
   if (data && data.fields && data.fields.length > 0) {
     for (let index = 0; index < data.size; index++) {
       const hit: { [key: string]: any } = {};
+
+      const processField = (field: any, value: any): any => {
+        // Handle date fields
+        if (moment(value, ['YYYY-MM-DD HH:mm:ss'], true).isValid()) {
```

Comment:
i can't remember this is before we know it is a date field correct? or do we already know it's a date field? as in field.type === 'date'.

`YYYY-MM-DD HH:mm:ss` seems too specific to PPL, correct me if im wrong.

would it make sense to do it something like

```
const tzConfig = opensearchDashboards.services.uiSettings!.get('dateFormat:tz');
const momentParsedValue = moment(value).tz(tzConfig);
if (momentParsedValue.isValid()) return momentParsedValue?.format('YYYY-MM-DDTHH:mm:ss.SSSZ');
```

or does it need to be +00:00 because to me it feels like we would be forcing any date value of `YYYY-MM-DD HH:mm:ss` to have a value we statically append `+00:00`

Reply:
yea at first i used field.type === 'date' but then i realized this will neglect those nested field which does not have a field.type yet, therefore i switch to use moment.

Reply:
We do not need to get the timezone here because opensearch backend always store time value as UTC, no matter which timezone we define in the advanced setting. So here we should treat the string we got back as in UTC time. 

We could potentially do: 
```
if (moment.utc(value).isValid()) {
          return moment.utc(value).format('YYYY-MM-DDTHH:mm:ss.SSSZ');
        }
```
but this doesnt work for some reason: when the value type is object, it will always go into the if loop as it is a date.

Reply:
i changed `return value.replace(' ', 'T') + '+00:00';` to `return moment.utc(value).format('YYYY-MM-DDTHH:mm:ssZ');` so it it more general